contrib/test/test_firedancer_testnet.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set -euxo pipefail
+shopt -s extglob
+IFS=$'\n\t'
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+FD_DIR="$SCRIPT_DIR/../.."
+
+# create temporary files in the user's home directory because it's likely to be on a large disk
+TMPDIR=$(mktemp --directory --tmpdir="$HOME" tmp-test-tvu-testnet.XXXXXX)
+cd $TMPDIR
+
+cleanup() {
+  sudo killall fddev || true
+  fddev configure fini all >/dev/null 2>&1 || true
+  rm -rf "$TMPDIR"
+}
+
+download_snapshot() {
+  local url=$1
+  local num_tries=${2:-10}
+  local s
+  for i in $(seq 1 $num_tries); do
+    s=$(curl -s --max-redirs 0 $url)
+    if ! wget -q --trust-server-names $url; then
+      sleep 1
+    else
+      echo "${s:1}"
+      return 0
+    fi
+  done
+
+  echo "failed after $num_tries tries to wget $url"
+  return 1
+}
+
+trap cleanup EXIT SIGINT SIGTERM
+sudo killall fddev || true
+
+# if fddev is not on path then use the one in the home directory
+if ! command -v fddev > /dev/null; then
+  PATH="$FD_DIR/build/native/gcc/bin":$PATH
+fi
+
+ENTRYPOINT=emam-ccn-solana-testnet-val1.jumpisolated.com
```

Comment:
this should be the public entrypoint by default. people outside cannot use this script.